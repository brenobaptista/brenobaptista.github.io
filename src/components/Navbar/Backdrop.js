import React from 'react';
import PropTypes from 'prop-types';

const backdrop = (props) => {
  const { show, clicked } = props;

  return (
    show ? (
      <div
        className="backdrop"
        onClick={clicked}
        onKeyPress={clicked}
        role="button"
        tabIndex="0"
        aria-label="backdrop"
      />
    ) : null
  );
};

backdrop.propTypes = {
  show: PropTypes.bool,
  clicked: PropTypes.func,
};

backdrop.defaultProps = {
  show: false,
  clicked: () => {},
};

export default backdrop;
