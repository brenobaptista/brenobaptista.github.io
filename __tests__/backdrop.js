import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import Backdrop from '../src/components/Navbar/Backdrop'

test('calls "clicked" prop on click', () => {
  const clicked = jest.fn()
  const { getByLabelText } = render(<Backdrop show clicked={clicked} />)

  fireEvent.click(getByLabelText('backdrop'))
  expect(clicked).toHaveBeenCalled()
})

test('displays nothing if "show" prop is false', () => {
  const { asFragment } = render(<Backdrop />)

  expect(asFragment(<Backdrop />)).toMatchSnapshot()
})
