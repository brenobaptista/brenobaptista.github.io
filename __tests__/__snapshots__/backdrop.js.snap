// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays nothing if "show" prop is false 1`] = `<DocumentFragment />`;
